src/brod_producers_sup.erl

```diff
@@ -55,6 +56,13 @@
 start_link(ClientPid, Producers) ->
   supervisor3:start_link(?MODULE, {?SUP, ClientPid, Producers}).
 
+%% @doc Dynamically start a per-topic supervisor
+-spec start_producer(pid(), pid(), topic(), producer_config()) ->
+                        {ok, pid()} | {error, any()}.
+start_producer(SupPid, ClientPid, TopicName, Config) ->
```

Comment:
we need a existence check on the topic name before starting the child.
or change it not to auto restart child on post-init failure.